CRM/Utils/Array.php

```diff
@@ -1076,4 +1076,23 @@ public static function convertCheckboxFormatToArray($input) {
     return $input;
   }
 
+  /**
+   * Ensure that array is encoded in utf8 format.
+   *
+   * @param array $array
+   *
+   * @return array $array utf8-encoded.
+   */
+  public static function encode_items($array) {
+    foreach ($array as $key => $value) {
+      if (is_array($value)) {
+        $array[$key] = self::encode_items($value);
+      }
+      else {
+        $array[$key] = mb_convert_encoding($value, mb_detect_encoding($value, mb_detect_order(), TRUE), 'UTF-8');
```

Comment:
OK, this has the effect of changing how some return values are encoded in SOAP XML -- ie as `xsd:int` vs `xsd:string`. A quick fix seems to be:

```
diff --git a/CRM/Utils/Array.php b/CRM/Utils/Array.php
index 8537409..eaca889 100644
--- a/CRM/Utils/Array.php
+++ b/CRM/Utils/Array.php
@@ -1088,6 +1088,9 @@ class CRM_Utils_Array {
       if (is_array($value)) {
         $array[$key] = self::encode_items($value);
       }
+      elseif (is_int($value)) {
+        $array[$key] = $value;
+      }
       else {
         $array[$key] = mb_convert_encoding($value, mb_detect_encoding($value, mb_detect_order(), TRUE), 'UTF-8');
       }
```